superset/hive_query.py

```diff
@@ -39,72 +34,123 @@
     'P1Y'  : 31556926
 }
 
-# return seconds from gran valaue map
-def getGranValueInSeconds(value):
-    if  value in GRAN_VALUE_MAP:
+# return seconds from gran valaue map default is 86400
+def get_gran_value_in_seconds(value,time_partitions):
+    partition_min_grain = get_partitions_min_grain(time_partitions)
+    if value in GRAN_VALUE_MAP and  GRAN_VALUE_MAP[value] >= partition_min_grain:
         return GRAN_VALUE_MAP[value]
+    return partition_min_grain
+
+def get_partitions_min_grain(time_partitions):
+    partition_grains = list()
+    if 'year' in time_partitions:
+        partition_grains.append(31556926)
+    if 'month' in time_partitions:
+        partition_grains.append(2629743)
+    if 'day' in time_partitions:
+        partition_grains.append(86400)
+    if 'hour' in time_partitions:
+        partition_grains.append(3600)
+    if 'minute' in time_partitions:
+        partition_grains.append(60)
+
+    partition_grains.sort()
```

Comment:
Do we really need sort. Aren't list is ordered already?

Reply:
we are appending items and it can be in any order and we nee min value so sorting it.

Reply:
done